libflux/src/semantic/tests.rs

```diff
@@ -74,19 +73,18 @@ fn parse_program(src: &str) -> ast::Package {
 ///
 macro_rules! test_infer {
     (
-        $(env: $env:expr,)?
+        env: $env:expr,
```

Comment:
This should be optional. Can you just move `m` inside the optional block?

Reply:
It is optional. You can specify multiple patterns. This is the pattern with the implementation. I had to do it this way because there didn't seem to be a way to say "if this pattern isn't included, do this instead." So instead, there's one pattern which includes it and one pattern that doesn't. The pattern that doesn't constructs an empty hashmap and passes that in.